File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/SearchCandidateAdapter.java
package com.workforcex.employer.ui.employer;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.workforcex.employer.R;
import com.workforcex.employer.api.CandidateSearchResult;
import java.util.List;

public class SearchCandidateAdapter extends RecyclerView.Adapter<SearchCandidateAdapter.ViewHolder> {

    private List<CandidateSearchResult> candidates;

    public SearchCandidateAdapter(List<CandidateSearchResult> candidates) {
        this.candidates = candidates;
    }

    public void update(List<CandidateSearchResult> newList) {
        this.candidates = newList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_search_candidate, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder h, int position) {
        CandidateSearchResult c = candidates.get(position);

        h.tvName.setText((position + 1) + ". " + (c.name != null ? c.name : c.mobileNumber));
        h.tvTotalScore.setText(String.format("%.0f%%", c.totalScore));
        h.tvSkills.setText("Skills: " + (c.skills != null ? c.skills : "—"));

        String details = "";
        if (c.experience != null) details += c.experience + " yrs exp";
        if (c.city != null)       details += (details.isEmpty() ? "" : " · ") + c.city;
        if (c.preferredSalary != null) details += "  ₹" + c.preferredSalary.intValue();
        h.tvDetails.setText(details);

        h.tvSkillScore.setText(String.format("%.0f%%", c.skillScore));
        h.tvExpScore.setText(String.format("%.0f%%", c.experienceScore));
        h.tvLocScore.setText(String.format("%.0f%%", c.locationScore));
        h.tvSalScore.setText(String.format("%.0f%%", c.salaryScore));
    }

    @Override
    public int getItemCount() { return candidates.size(); }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvName, tvTotalScore, tvSkills, tvDetails;
        TextView tvSkillScore, tvExpScore, tvLocScore, tvSalScore;

        ViewHolder(View v) {
            super(v);
            tvName       = v.findViewById(R.id.tvName);
            tvTotalScore = v.findViewById(R.id.tvTotalScore);
            tvSkills     = v.findViewById(R.id.tvSkills);
            tvDetails    = v.findViewById(R.id.tvDetails);
            tvSkillScore = v.findViewById(R.id.tvSkillScore);
            tvExpScore   = v.findViewById(R.id.tvExpScore);
            tvLocScore   = v.findViewById(R.id.tvLocScore);
            tvSalScore   = v.findViewById(R.id.tvSalScore);
        }
    }
}

File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/SearchCandidateAdapter.kt
package com.workforcex.employer.ui.employer

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.workforcex.employer.R
import com.workforcex.shared_employer.models.CandidateSearchResult

class SearchCandidateAdapter(private var candidates: List<CandidateSearchResult>) :
    RecyclerView.Adapter<SearchCandidateAdapter.ViewHolder>() {

    fun update(newList: List<CandidateSearchResult>) {
        this.candidates = newList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_search_candidate, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(h: ViewHolder, position: Int) {
        val c = candidates[position]

        h.tvName.text = (position + 1).toString() + ". " + (c.name ?: c.mobileNumber)
        h.tvTotalScore.text = String.format("%.0f%%", c.totalScore)
        h.tvSkills.text = "Skills: " + (c.skills ?: "—")

        var details = ""
        if (c.experience != null) details += c.experience.toString() + " yrs exp"
        if (c.city != null) details += (if (details.isEmpty()) "" else " · ") + c.city
        if (c.preferredSalary != null) details += "  ₹" + c.preferredSalary.toInt()
        h.tvDetails.text = details

        h.tvSkillScore.text = String.format("%.0f%%", c.skillScore)
        h.tvExpScore.text = String.format("%.0f%%", c.experienceScore)
        h.tvLocScore.text = String.format("%.0f%%", c.locationScore)
        h.tvSalScore.text = String.format("%.0f%%", c.salaryScore)

        h.itemView.setOnClickListener {
            val intent = Intent(h.itemView.context, CandidateProfileActivity::class.java)
            intent.putExtra("workerId", c.workerId)
            h.itemView.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int = candidates.size

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        val tvName: TextView = v.findViewById(R.id.tvName)
        val tvTotalScore: TextView = v.findViewById(R.id.tvTotalScore)
        val tvSkills: TextView = v.findViewById(R.id.tvSkills)
        val tvDetails: TextView = v.findViewById(R.id.tvDetails)
        val tvSkillScore: TextView = v.findViewById(R.id.tvSkillScore)
        val tvExpScore: TextView = v.findViewById(R.id.tvExpScore)
        val tvLocScore: TextView = v.findViewById(R.id.tvLocScore)
        val tvSalScore: TextView = v.findViewById(R.id.tvSalScore)
    }
}